src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -1651,6 +1651,11 @@ protected virtual void VisitFinallyBlock(BoundStatement finallyBlock, ref TLocal
             VisitStatement(finallyBlock); // this should generate no pending branches
         }
 
+        public override BoundNode VisitExtractedFinallyBlock(BoundExtractedFinallyBlock node)
```

Comment:
>BoundExtractedFinallyBlock [](start = 61, length = 26)

Doesn't this node only existing in lowering? If so, this visit method should throw an exception.

Reply:
Async rewriting uses data flow analysis (in `StateMachineRewriter.Rewrite` at line 106, we do `IteratorAndAsyncCaptureWalker.Analyze` on the incoming `body`). At that stage (after local rewriting and async exception handler rewriting), we have extracted finally blocks.

---
In reply to: [240715273](https://github.com/dotnet/roslyn/pull/31527#discussion_r240715273) [](ancestors = 240715273)